src/main/java/lotto/Rank.java

```diff
@@ -0,0 +1,37 @@
+package lotto;
+
+import java.util.Arrays;
+
+public enum Rank {
+    FIRST(6, 2000000000),
+    SECOND(5, 1500000),
+    THIRD(4, 50000),
+    FOURTH(3, 5000),
+    NO_MATCH(0, 0);
+
+    private int matchCount;
+    private int money;
+
+    Rank(int matchCount, int money) {
+        this.matchCount = matchCount;
+        this.money = money;
+    }
+
+
+    public static Rank rank(int matchCount) {
+        if(matchCount < FOURTH.matchCount) {
+            return NO_MATCH;
+        }
+        return Arrays.stream(values())
+                .filter(rank -> rank.matchCount == matchCount)
+                .findFirst()
+                .orElseThrow(IllegalAccessError::new);
```

Comment:
```suggestion
        return Arrays.stream(values())
                .filter(rank -> rank.matchCount == matchCount)
                .findFirst()
                .orElse(NO_MATCH);
```

지금의 구조는 Rank의 조건이 변경된다면 해당 메서드를 신경써줘야 하는 문제가 있네요!
위와 같이 개선하여 최대한 관리 포인트를 줄여볼 수 있을 것 같습니다 :)

Reply:
네. Rank의 조건이 변경되면 if문도 바꿔야하는 번거로움이 있을 것 같습니다. 힌트 반영해서 관리 포인트를 줄여보도록 리팩토링했습니다.